core/variant_op.cpp

```diff
@@ -161,35 +162,78 @@ bool Variant::booleanize(bool &r_valid) const {
 			return _data._real;
 		case STRING:
 			return (*reinterpret_cast<const String *>(_data._mem)) != "";
-		case VECTOR2:
-		case RECT2:
-		case TRANSFORM2D:
-		case VECTOR3:
-		case PLANE:
-		case RECT3:
-		case QUAT:
-		case BASIS:
-		case TRANSFORM:
-		case COLOR:
+		case VECTOR2: {
+			const Vector2 *v = reinterpret_cast<const Vector2 *>(_data._mem);
+			return v->x + v->y;
+		}
+		case RECT2: {
+			const Rect2 *v = reinterpret_cast<const Rect2 *>(_data._mem);
+			return v->position.x + v->position.y + v->size.x + v->size.y;
+		}
+		case TRANSFORM2D: {
+			const Transform2D *v = _data._transform2d;
+			return v->elements[0].x + v->elements[0].y +
+				   v->elements[1].x + v->elements[1].y +
+				   v->elements[2].x + v->elements[2].y;
+		}
+		case VECTOR3: {
+			const Vector3 *v = reinterpret_cast<const Vector3 *>(_data._mem);
+			return v->x + v->y + v->z;
+		}
+		case PLANE: {
+			const Plane *v = reinterpret_cast<const Plane *>(_data._mem);
+			return v->d + v->normal.x + v->normal.y + v->normal.z;
+		}
+		case RECT3: {
+			const Rect3 *v = _data._rect3;
+			return v->position.x + v->position.y + v->position.z +
+				   v->size.x + v->size.y + v->size.z;
+		}
+		case QUAT: {
+			const Quat *v = reinterpret_cast<const Quat *>(_data._mem);
+			return v->w + v->x + v->y + v->z;
+		}
+		case BASIS: {
+			const Basis *v = _data._basis;
+			return v->elements[0].x + v->elements[0].y + v->elements[0].z +
+				   v->elements[1].x + v->elements[1].y + v->elements[1].z +
+				   v->elements[2].x + v->elements[2].y + v->elements[2].z;
+		}
+		case TRANSFORM: {
+			const Transform *v = _data._transform;
+			return v->basis.elements[0].x + v->basis.elements[0].y + v->basis.elements[0].z +
+				   v->basis.elements[1].x + v->basis.elements[1].y + v->basis.elements[1].z +
+				   v->basis.elements[2].x + v->basis.elements[2].y + v->basis.elements[2].z +
+				   v->origin.x + v->origin.y + v->origin.z;
+		}
+		case COLOR: {
+			const Color *v = reinterpret_cast<const Color *>(_data._mem);
+			return v->a + v->b + v->g + v->r;
+		}
 		case _RID:
 			return (*reinterpret_cast<const RID *>(_data._mem)).is_valid();
 		case OBJECT:
 			return _get_obj().obj;
 		case NODE_PATH:
 			return (*reinterpret_cast<const NodePath *>(_data._mem)) != NodePath();
 		case DICTIONARY:
+			return !(*reinterpret_cast<const Dictionary *>(_data._mem)).empty();
 		case ARRAY:
+			return !(*reinterpret_cast<const Array *>(_data._mem)).empty();
```

Comment:
For the other *Array types you used `*Array.size()`, is there a difference in cost using `!Array.empty()` here?